integrationTests/vm/txsFee/relayedScCalls_test.go

```diff
@@ -19,202 +19,245 @@ import (
 )
 
 func TestRelayedScCallShouldWork(t *testing.T) {
-	testContext, err := vm.CreatePreparedTxProcessorWithVMs(config.EnableEpochs{
-		DynamicGasCostForDataTrieStorageLoadEnableEpoch: integrationTests.UnreachableEpoch,
-	})
-	require.Nil(t, err)
-	defer testContext.Close()
+	t.Run("before relayed fix", testRelayedScCallShouldWork(integrationTests.UnreachableEpoch))
+	t.Run("after relayed fix", testRelayedScCallShouldWork(0))
+}
 
-	scAddress, _ := utils.DoDeploy(t, testContext, "../wasm/testdata/counter/output/counter.wasm")
-	utils.CleanAccumulatedIntermediateTransactions(t, testContext)
+func testRelayedScCallShouldWork(relayedFixActivationEpoch uint32) func(t *testing.T) {
+	return func(t *testing.T) {
+		testContext, err := vm.CreatePreparedTxProcessorWithVMs(config.EnableEpochs{
+			DynamicGasCostForDataTrieStorageLoadEnableEpoch: integrationTests.UnreachableEpoch,
+			FixRelayedMoveBalanceEnableEpoch:                relayedFixActivationEpoch,
+		})
+		require.Nil(t, err)
+		defer testContext.Close()
 
-	relayerAddr := []byte("12345678901234567890123456789033")
-	sndAddr := []byte("12345678901234567890123456789112")
-	gasLimit := uint64(1000)
+		scAddress, _ := utils.DoDeploy(t, testContext, "../wasm/testdata/counter/output/counter.wasm")
+		utils.CleanAccumulatedIntermediateTransactions(t, testContext)
 
-	_, _ = vm.CreateAccount(testContext.Accounts, sndAddr, 0, big.NewInt(0))
-	_, _ = vm.CreateAccount(testContext.Accounts, relayerAddr, 0, big.NewInt(30000))
+		relayerAddr := []byte("12345678901234567890123456789033")
+		sndAddr := []byte("12345678901234567890123456789112")
+		gasLimit := uint64(1000)
 
-	userTx := vm.CreateTransaction(0, big.NewInt(100), sndAddr, scAddress, gasPrice, gasLimit, []byte("increment"))
+		_, _ = vm.CreateAccount(testContext.Accounts, sndAddr, 0, big.NewInt(0))
+		_, _ = vm.CreateAccount(testContext.Accounts, relayerAddr, 0, big.NewInt(30000))
 
-	rtxData := integrationTests.PrepareRelayedTxDataV1(userTx)
-	rTxGasLimit := 1 + gasLimit + uint64(len(rtxData))
-	rtx := vm.CreateTransaction(0, userTx.Value, relayerAddr, sndAddr, gasPrice, rTxGasLimit, rtxData)
+		userTx := vm.CreateTransaction(0, big.NewInt(100), sndAddr, scAddress, gasPrice, gasLimit, []byte("increment"))
 
-	retCode, err := testContext.TxProcessor.ProcessTransaction(rtx)
-	require.Equal(t, vmcommon.Ok, retCode)
-	require.Nil(t, err)
+		rtxData := integrationTests.PrepareRelayedTxDataV1(userTx)
+		rTxGasLimit := 1 + gasLimit + uint64(len(rtxData))
```

Comment:
can you replace the 1 with a variable instead?

also valid in other places in this file